lib/app/server.js

```diff
@@ -81,7 +81,7 @@ export default context => {
     // nuxtServerInit
     <% if (store) { %>
       let promise = (store._actions && store._actions.nuxtServerInit ? store.dispatch('nuxtServerInit', omit(getContext(context), 'redirect', 'error')) : null)
-      if (!(promise instanceof Promise)) promise = Promise.resolve()
+      if (!promise || (!(promise instanceof Promise) && (promise && typeof promise.then !== 'function'))) promise = Promise.resolve()
```

Comment:
I believe the if can be simplify to this:
```js
if (!promise || (!(promise instanceof Promise) && typeof promise.then !== 'function'))
```

If `promise` does not pass the first check (`!promise`) it won't check the 2nd check.

Can you update this please?